collections/max_by.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * Returns the first element that is the largest value of the given function or undefined if there are no elements.
+ *
+ * Example:
+ *
+ * ```ts
+ * import { maxBy } from "./max_by.ts";
+ * import { assertEquals } from "../testing/asserts.ts";
+ *
+ * const people = [
+ *     { name: 'Anna', age: 34 },
+ *     { name: 'Kim', age: 42 },
+ *     { name: 'John', age: 23 },
+ * ];
+ *
+ * const personWithMaxAge = maxBy(people, i => i.age);
+ *
+ * assertEquals(personWithMaxAge, { name: 'Kim', age: 42 });
+ * ```
+ */
+export function maxBy<T>(
+  collection: Array<T>,
```

Comment:
I think we have settled on `T[]` over `Array<T>` in the module now. Also mark as `readonly` to ensure no mutation happens:

```suggestion
  readonly collection: T[],
```